File: services/frontend/src/lib/api/schools/get-school-by-id.ts
import { convertStructTypes } from "$lib/functions/convertStructTypes";
import type { SchoolModelType } from ".";

export async function getSchoolByID(schoolID: string): Promise<SchoolModelType | undefined> {
    try {
        const response = await fetch(`/api/v1/schools/${schoolID}`);
        if (!response.ok) {
            throw new Error('Failed to fetch school data by id');
        }

        const rawData = await response.json()
        const data = convertStructTypes(rawData)

        return data;
    } catch (error) {

        console.error('Failed to fetch school data by id', error);
        return undefined
    }
}

File: services/backend/src/resources/schools/http/get_school_by_id.go
package http

import (
	"net/http"

	customerrorconst "github.com/dcssoftware/bafoeg-manager/src/helper/debug/customerrors/custom-error-const"
	uuidvalidator "github.com/dcssoftware/bafoeg-manager/src/helper/uuid-validator"
	"github.com/dcssoftware/bafoeg-manager/src/resources/schools/http/models"
	"github.com/gofiber/fiber/v3"
)

//	@Summary		V1 Get school by ID
//	@Description	Retrieves V1 School by ID
//	@Tags			schools
//	@Produce		json
//	@Param			schoolID	path		string	true	"School ID"
//
// Success 		200 {object} 	models.SchoolModel "School Model"
//
//	@Failure		404			{string}	string	"The requested data could not be found."
//	@Failure		500			{string}	string	"An error occurred while processing your request. Please try again later. "
//
//	@Router			/api/v1/schools/{schoolID} [get]
func (h *SchoolHandler) GetSchoolByID(c fiber.Ctx) error {

	schoolIDString := c.Params("schoolID", "")
	if schoolIDString == "" || !uuidvalidator.ValidateUUID(schoolIDString) {
		return c.Status(http.StatusBadRequest).SendString(customerrorconst.BAD_REQUEST_ERROR_MESSAGE)
	}

	schools, schoolsErr := h.service.GetSchoolByID(schoolIDString)
	if schoolsErr != nil {
		status, message := schoolsErr.HTTPError()
		return c.Status(status).SendString(message)
	}

	httpSchools := models.ToHttpSchoolModel(schools)
	return c.Status(http.StatusOK).JSON(httpSchools)
}
